devicetypes/smartthings/zigbee-metering-dimmer.src/zigbee-metering-dimmer.groovy

```diff
@@ -0,0 +1,142 @@
+/**
+ *  Copyright 2021 SmartThings
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+import physicalgraph.zigbee.zcl.DataType
+metadata {
+	definition (name: "ZigBee Metering Dimmer", namespace: "smartthings", author: "SmartThings", ocfDeviceType: "oic.d.light",  mnmn: "SmartThings", vid:"generic-dimmer-power-energy", runLocally: true, executeCommandsLocally: true, genericHandler: "Zigbee", minHubCoreVersion: '000.019.00012') {
+
+		capability "Actuator"
+		capability "Configuration"
+		capability "Refresh"
+		capability "Power Meter"
+		capability "Energy Meter"
+		capability "Switch"
+		capability "Switch Level"
+		capability "Health Check"
+
+		// Enbrighten/Jasco
+		fingerprint manufacturer: "Jasco Products", model: "43082", deviceJoinName: "Enbrighten Dimmer Switch" //Enbrighten, in-Wall Smart Dimmer With Energy Monitoring 43082, Raw Description: 01 0104 0101 00 08 0000 0003 0004 0005 0006 0008 0702 0B05 02 000A 0019
+	}
+}
+
+def getATTRIBUTE_READING_INFO_SET() { 0x0000 }
+def getATTRIBUTE_HISTORICAL_CONSUMPTION() { 0x0400 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+	log.debug "description is $description"
+	List result = []
+
+	def event = zigbee.getEvent(description)
+	if (event) {
+		log.info event
+		if (event.name == "power") {
+			def powerDiv = device.getDataValue("powerDivisor")
+			powerDiv = powerDiv ? (powerDiv as int) : 1
+			event.value = event.value/powerDiv
+			event.unit = "W"
+		} else if (event.name == "energy") {
+			def energyDiv = device.getDataValue("energyDivisor")
+			energyDiv = energyDiv ? (energyDiv as int) : 100
+			event.value = event.value/energyDiv
+			event.unit = "kWh"
+		}
+		log.info "event: $event"
+		result << event
+	} else {
+		def descMap = zigbee.parseDescriptionAsMap(description)
+		log.debug "descMap: $descMap"
+
+		List attrData = [[clusterInt: descMap.clusterInt ,attrInt: descMap.attrInt, value: descMap.value]]
+		descMap.additionalAttrs.each {
+			attrData << [clusterInt: descMap.clusterInt, attrInt: it.attrInt, value: it.value]
+		}
+
+		attrData.each {
+			def map = [:]
+			if (it.value && it.clusterInt == zigbee.SIMPLE_METERING_CLUSTER && it.attrInt == ATTRIBUTE_HISTORICAL_CONSUMPTION) {
+				log.debug "power"
+				map.name = "power"
+				def powerDiv = device.getDataValue("powerDivisor")
```

Comment:
so in order for this to work with LE, the data value has to be named "divisor"

LE is only for power, so energy can still be whatever you want

Reply:
ok, I just thought it had to kept as state variable with any name. Fixed.